src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -408,9 +498,120 @@ case class WorkflowActor(workflow: WorkflowDescriptor,
     // This only does the initialization for a newly created workflow.  For a restarted workflow we should be able
     // to assume the adjusted symbols already exist in the DB, but is it safe to assume the staged files are in place?
     val adjustedInputs = backend.initializeForWorkflow(workflow)
-    dataAccess.createWorkflow(workflowInfo, buildSymbolStoreEntries(workflow.namespace, adjustedInputs), workflow.namespace.workflow.calls, backend)
+    dataAccess.createWorkflow(workflowInfo, buildSymbolStoreEntries(workflow.namespace, adjustedInputs), workflow.namespace.workflow.children, backend)
+  }
+
+  /**
+   * This is the lookup function used to evaluate scatter collection expressions.
+   *
+   * For example, scatter(x in foo.bar) would evaluate the collection "foo.bar"
+   * and call this lookup function on "foo".
+   *
+   * This implementation takes a few shortcuts and tries to find a Call or
+   * Declaration with the given name in the workflow.
+   *
+   * A more long-term approach would be to traverse the scope hierarchy to resolve a variable into
+   * the closest definition in scope.
+   */
+  private def scatterCollectionLookupFunction(workflow: Workflow, key: ExecutionStoreKey)(identifier: String): WdlValue = {
+    resolveIdentifierOrElse(identifier, lookupCall(key, workflow), lookupDeclaration(workflow)) {
+      throw new WdlExpressionException(s"Could not resolve identifier '$identifier' as a call or declaration.")
+    }
+  }
+
+  private def isWorkflowDone: Boolean = executionStore forall isDone
+
+  private def isWorkflowAborted: Boolean = executionStore forall { x => isTerminal(x._2) || x._2 == ExecutionStatus.NotStarted }
```

Comment:
You can get rid of the `x._2` business with `executionStore values`, but I'm not sure why this logic would correspond to the workflow being aborted